pkg/pool-weighted/contracts/smart/ManagedPool.sol

```diff
@@ -326,6 +327,89 @@ contract ManagedPool is BaseWeightedPool, ProtocolFeeCache, ReentrancyGuard {
         emit SwapEnabledSet(swapEnabled);
     }
 
+    /**
+     * @dev Removes a token from the Pool's composition, withdrawing all funds from the Vault and sending them to
+     * `recipient`, and adjusting the weights of all other tokens.
+     *
+     * Tokens can only be removed if the Pool has more than 2 tokens, as it can never have fewer than 2. Token removal
+     * is also forbidden during a weight change, or if one is scheduled to happen in the future.
+     *
+     * The caller may aditionally pass a non-zero `burnAmount` to have some of their BPT be burned, which might be
+     * useful in some scenarios to account for the fact that the Pool now has fewer tokens.
+     */
+    function removeToken(
+        IERC20 token,
+        address recipient,
+        uint256 burnAmount
+    ) external authenticate nonReentrant whenNotPaused {
+        // Exit the pool, returning the full balance of the token to the recipient
+        (IERC20[] memory tokens, uint256[] memory unscaledBalances, ) = getVault().getPoolTokens(getPoolId());
+        _require(tokens.length > 2, Errors.MIN_TOKENS);
+
+        // Tokens cannot be removed during or before a weight change to reduce complexity of weight interactions
+        _ensureNoWeightChange();
+
+        // Reverts if token does not exist in pool.
+        uint256 tokenIndex = _tokenAddressToIndex(tokens, token);
+        uint256 tokenBalance = unscaledBalances[tokenIndex];
+        uint256 tokenNormalizedWeight = _getNormalizedWeight(token);
+
+        // We first perform a special exit operation at the Vault, which will withdraw the entire tokenBalance from it.
+        // Only the Pool itself is authorized to initiate such an exit.
+        uint256[] memory minAmountsOut = new uint256[](tokens.length);
+        minAmountsOut[tokenIndex] = tokenBalance;
+
+        getVault().exitPool(
+            getPoolId(),
+            address(this),
+            payable(recipient),
+            IVault.ExitPoolRequest({
+                assets: _asIAsset(tokens),
+                minAmountsOut: minAmountsOut,
+                userData: abi.encode(WeightedPoolUserData.ExitKind.REMOVE_TOKEN, tokenIndex),
+                toInternalBalance: false
+            })
+        );
```

Comment:
```suggestion
        getVault().exitPool(
            getPoolId(),
            address(this),
            payable(recipient),
            IVault.ExitPoolRequest({
                assets: _asIAsset(tokens),
                minAmountsOut: new uint256[](tokens.length),
                userData: abi.encode(WeightedPoolUserData.ExitKind.REMOVE_TOKEN, tokenIndex),
                toInternalBalance: false
            })
        );
```

L360 doesn't actually do anything as the pools balances would have been manipulated before we enter this function (and so before we read `tokenBalance`).

Reply:
Addressed below

Reply:
Marking this as resolved as this will be addressed by https://github.com/balancer-labs/balancer-v2-monorepo/pull/1108#discussion_r857824669